community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java

```diff
@@ -185,11 +186,15 @@ private void verifyRecordFormat()
         try
         {
             String expectedStoreVersion = recordFormats.storeVersion();
-            String actualStoreVersion = versionLongToString( getRecord( pageCache, neoStoreFileName, STORE_VERSION ) );
-            RecordFormats actualStoreFormat = RecordFormatSelector.selectForVersion( actualStoreVersion );
-            if ( !isCompatibleFormats( actualStoreFormat ) )
+            long record = getRecord( pageCache, neoStoreFileName, STORE_VERSION );
+            if ( record != MetaDataRecordFormat.FIELD_NOT_PRESENT )
```

Comment:
when it's the case that we do not have STORE_VERSION in metadata store? and why it's safe to skip format check in that case?